go/libraries/doltcore/sqle/dtables/column_diff_table.go

```diff
@@ -0,0 +1,646 @@
+	now := time.Now()
+	dp := NewDiffPartition(delta.ToTable, delta.FromTable, delta.ToName, delta.FromName, (*dtypes.Timestamp)(&now), (*dtypes.Timestamp)(&now), delta.ToSch, delta.FromSch)
+	ri := NewDiffPartitionRowIter(*dp, ddb, j)
+	colNames, diffTypes, err := calculateColDelta(ctx, ri, delta.ToSch.GetAllCols(), delta.FromSch.GetAllCols(), diffTableSchema.GetAllCols())
+	if err != nil {
+		return nil, err
+	}
+
+	return &tableColChange{
+		tableName: delta.ToName,
+		colNames:  colNames,
+		diffTypes: diffTypes,
+	}, nil
+}
+
+// calculateColDelta iterates through the rows of the given table delta and compares each cell in the to_ and from_
+// cells to compile a list of modified columns
+func calculateColDelta(ctx *sql.Context, iter sql.RowIter, toCols, fromCols, diffTableCols *schema.ColCollection) ([]string, []string, error) {
+	colNamesSet := make(map[string]struct{})
+	var resultColNames []string
+	var resultDiffTypes []string
+	modifiedCols, addedCols, droppedCols, _, _ := calculateColSchemaDiff(toCols, fromCols)
+
+	// add all added/dropped columns to result
+	for _, col := range addedCols {
+		resultColNames = append(resultColNames, col)
+		resultDiffTypes = append(resultDiffTypes, diffTypeAdded)
+	}
+	for _, col := range droppedCols {
+		resultColNames = append(resultColNames, col)
+		resultDiffTypes = append(resultDiffTypes, diffTypeRemoved)
+	}
+
+	// check each row for diffs in modified columns
+	for {
+		r, err := iter.Next(ctx)
+		if err == io.EOF {
+			for col := range colNamesSet {
+				// append modified columns to result
+				resultColNames = append(resultColNames, col)
+				resultDiffTypes = append(resultDiffTypes, diffTypeModified)
+			}
+			return resultColNames, resultDiffTypes, nil
+		} else if err != nil {
+			return nil, nil, err
+		}
+
+		// only need to check modified columns
+		for _, col := range modifiedCols {
+			toColTag := diffTableCols.NameToCol["to_"+col].Tag
+			fromColTag := diffTableCols.NameToCol["from_"+col].Tag
+			toIdx := diffTableCols.TagToIdx[toColTag]
+			fromIdx := diffTableCols.TagToIdx[fromColTag]
+
+			if r[toIdx] != r[fromIdx] {
+				colNamesSet[col] = struct{}{}
+			}
+		}
+
+		// can stop checking rows when we already have all modified columns in the result set
+		if len(colNamesSet) == len(modifiedCols) {
+			for col := range colNamesSet {
+				// append modified columns to result
+				resultColNames = append(resultColNames, col)
+				resultDiffTypes = append(resultDiffTypes, diffTypeModified)
+			}
+			return resultColNames, resultDiffTypes, nil
+		}
+	}
+}
+
+// calculateColSchemaDiff calculates which columns were modified, added, or dropped between to and from schemas and
+// returns a list of column names for each type of change, the total list of column names, and a corresponding list of
+// diff_types for each column
+func calculateColSchemaDiff(toCols *schema.ColCollection, fromCols *schema.ColCollection) ([]string, []string, []string, []string, []string) {
```

Comment:
Great godocs! It's an important one to inform about which returned value corresponds to with column array especially if there are many returned values of the same type. (small suggestion, but up to your preference 🙂) you can also make small struct with added, dropped, modified and all columns array of strings to return, then it can be easily accessed with each members.